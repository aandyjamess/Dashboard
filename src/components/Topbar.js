import React from "react";
import PropTypes from "prop-types";

import ConnectionStatus from "./ConnectionStatus";

const Topbar = ({connected}) => (
  <nav className="topbar navbar">
    <span className="icon icon-logo" />
    <ConnectionStatus connected={connected} />
    <span className="topbar-avatar icon icon-user_selfie_ph" />
  </nav>
);

Topbar.PropTypes = {
  connected: PropTypes.bool
};

Topbar.defaultProps = {
  connected: false
};

export default Topbar;
