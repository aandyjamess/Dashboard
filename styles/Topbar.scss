@import "./variables";

.topbar {
  display: flex;
  flex-direction: row;
  align-items: center;
  position: fixed;
  height: $topbar-height;
  left: 0;
  right: 0;
  top: 0;
  padding-left: $topbar-margin-left;
  padding-right: $topbar-margin-right;
  border-bottom: 1px solid $navbar-default-border;
  z-index: 2000;
  background-color: $navbar-default-bg;

  &-avatar {
    border: 1px solid grey;
    border-radius: 50%;
  }
}
