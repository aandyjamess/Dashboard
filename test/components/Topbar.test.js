import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Topbar from "../../src/components/Topbar";

describe("Topbar", () => {
  it("renders correctly", () => {
    expect(toJson(shallow(<Topbar connected={false} />))).toMatchSnapshot();
  });
});
