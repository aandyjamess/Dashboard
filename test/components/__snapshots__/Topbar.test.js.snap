// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar renders correctly 1`] = `
<nav
  className="topbar navbar"
>
  <span
    className="icon icon-logo"
  />
  <ConnectionStatus
    connected={false}
  />
  <span
    className="topbar-avatar icon icon-user_selfie_ph"
  />
</nav>
`;
